src/addMultiplier.js

```diff
@@ -8,7 +8,7 @@ const addMultiplier = (expression) => {
   const pAndp = /(\))(\()/g;
 
   //return the value
-  return expression
+  return (expression + "")
```

Comment:
@grimxreaper in the commit "extract handleButtonPress", is there a way you could make it so you don't need to sprinkle `(variable + "")` everywhere?

Reply:
I added an empty string to this expression here:
`  var cleanEquation = result + "";`
So that I could remove the `+ ""` in 2 other places. Basically I'm thinking along the lines of if I can change the variable into a string using that method or `.toString()` then it is altered earlier before it is passed in and manipulated anywhere else.

Reply:
Right! Changing it into a string earlier will allow you to assume it is a string anywhere else. I’m wondering if you can add some code comments to identify the earliest places you can make it into a string?

Reply:
When it is first passed in to `handleButtonPress`? Like so:
```
 const [nextResult, nextEquation] = handleButtonPress(
      result.toString(),
      button,
      lastEquation
    );

```

Reply:
This may be something you need to try running code to find.

Reply:
Yes sorry I had tried it, it works 👍🏼

Reply:
But did you remove the plus quote part from all the `(variable + “”)` when you tried it?

Reply:
To really test it, you need to first search your code and remove `+ “”`

Reply:
Yes I removed it from `expression` in `addMultiplier`, like in the example you gave. As well as from `cleanequation` in `handleButtonPress` in two places (aka everywhere that had that + ""). The only places it still needs those is in the `getLastChar` and `hasLastChar`.

Reply:
Is there a way we can make it so those two functions don’t need it either? What I’m hinting at is that the location you chose is not the earliest location. How does result end up as a number?

Reply:
Because of the numbers in the Keypad component right?

Reply:
Result turns from a string into a number in handleNestedParens, and when returned from runEquation too.

Reply:
Where did the keypad comment go?? That was a good answer. As well as handleNestedParens.

Reply:
Oh! I deleted it because I thought that that's not exactly where it turns into a number. Okay noted.

Reply:
Keypad is where it starts as a number! Could that change?